pkg/auth/prometheus/metrics.go

```diff
@@ -0,0 +1,36 @@
+package prometheus
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	AuthSubsystem = "auth_subsystem"
+)
+
+var (
+	authCounterTotal = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Subsystem: AuthSubsystem,
+			Name:      "auth_count_total",
+			Help:      "Counts total authentication attempts",
+		}, []string{},
+	)
+	authCounterResult = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Subsystem: AuthSubsystem,
+			Name:      "auth_count_result",
+			Help:      "Counts authentication attempts by result",
+		}, []string{"result"},
+	)
+)
+
+func init() {
+	prometheus.MustRegister(authCounterTotal)
+	prometheus.MustRegister(authCounterResult)
+}
+
+func UpdateAuthCounters(result string) {
```

Comment:
Generally we don't abstract metrics recording like this.  Call this method "record"